prov/tcp/src/tcpx_conn_mgr.c

```diff
@@ -338,7 +351,8 @@ static void server_recv_connreq(struct util_wait *wait,
 	handle  = container_of(cm_ctx->fid, struct tcpx_conn_handle, handle);
 
 	FI_DBG(&tcpx_prov, FI_LOG_EP_CTRL, "Server receive connect request\n");
-	ret = rx_cm_data(handle->sock, &conn_req, ofi_ctrl_connreq, cm_ctx);
+	ret = rx_cm_data(handle->sock, &conn_req, ofi_ctrl_connreq,
+			 ofi_ctrl_none, cm_ctx);
```

Comment:
I think I understand why you're trying to do with this.  The only valid value hdr type is ofi_ctrl_connreq, so nack should not occur.  And really, the only difference is whether a debug message is printed as a warning or info.  I think we just add a nack type check to rx_cm_data and avoid the extra complexity the results by defining ctrl_none.

Reply:
Ok, I guess you  want to have the simpler version I initially had. I thought it is a bit cleaner that way, although maybe I should have used a ptr. Well going back to the simplest version now.